apps/libsignpost/signpost_energy_monitors.c

```diff
@@ -0,0 +1,186 @@
+//Initialization functions
+//Make sure to call the right one
+/////////////////////////////////////////////////////////
+void signpost_energy_init_ltc2941 (void) {
+    // configure each ltc with the correct prescaler
+    for (int i = 0; i < 8; i++) {
+        i2c_selector_select_channels_sync(1<<i);
+        ltc294x_configure_sync(LTC2941, InterruptPinAlertMode, POWER_MODULE_PRESCALER_LTC2941, VbatAlertOff);
+    }
+
+    model = LTC2941;
+    prescaler = POWER_MODULE_PRESCALER_LTC2941;
+
+    // set all channels open for Alert Response
+    i2c_selector_select_channels_sync(0xFF);
+}
+
+void signpost_energy_init_ltc2943 (void) {
+    // configure each ltc with the correct prescaler
+    for (int i = 0; i < 9; i++) {
+        i2c_selector_select_channels_sync(1<<i);
+        ltc294x_configure_sync(LTC2943, InterruptPinAlertMode, POWER_MODULE_PRESCALER_LTC2943, ADCScan);
+    }
+
+    model = LTC2943;
+    prescaler = POWER_MODULE_PRESCALER_LTC2943;
+
+    // set all channels open for Alert Response
+    i2c_selector_select_channels_sync(0x1FF);
+
+    //max17205_configure_pack_sync();
+}
+
+////////////////////////////////////////////////
+// Static helper functions
+// ////////////////////////////////////////////
+static int get_ltc_energy_uah (int selector_mask, int rsense) {
+	// Select correct LTC294x
+	i2c_selector_select_channels_sync(selector_mask);
+
+    return ltc294x_convert_to_coulomb_uah(ltc294x_get_charge_sync(),rsense,prescaler,model);
+}
+
+static int get_ltc_current_ua (int selector_mask, int rsense) {
+	// Select correct LTC294x
+    i2c_selector_select_channels_sync(selector_mask);
+
+    return ltc294x_convert_to_current_ua(ltc294x_get_current_sync(),rsense);
+}
+
+static void reset_ltc_energy (int selector_mask) {
+	// Select correct LTC294x
+    i2c_selector_select_channels_sync(selector_mask);
+
+	ltc294x_reset_charge_sync();
+}
+
+///////////////////////////////////////////////////
+// Energy Reset Functions
+// ////////////////////////////////////////////////
+void signpost_energy_reset_all_energy (void) {
+    signpost_energy_reset_controller_energy();
+    signpost_energy_reset_solar_energy();
+    signpost_energy_reset_linux_energy();
+    for(uint8_t i = 0; i < 8; i++) {
+        signpost_energy_reset_module_energy(i);
+    }
+}
+
+void signpost_energy_reset_controller_energy (void) {
+    reset_ltc_energy(0x1);
+}
+
+void signpost_energy_reset_module_energy (int module_num) {
+    if(module_num != 3 && module_num != 4) {
+        reset_ltc_energy(module_num_to_selector_mask[module_num]);
+    }
+}
+
+void signpost_energy_reset_solar_energy (void) {
+    reset_ltc_energy(0x100);
+}
+
+void signpost_energy_reset_linux_energy (void) {
+    reset_ltc_energy(0x2);
+}
+
+///////////////////////////////////////////////////
+// Raw coulomb counter read functions
+// ///////////////////////////////////////////////
+uint32_t signpost_energy_get_controller_energy (void) {
```

Comment:
Why does this not have a unit attached to it any more? The interior function has `uah`, but we're hiding that from the caller? I very consciously put units on every function when I wrote this the first time, it's makes it way more usable
(edit: all of these need units)